networkx/generators/directed.py

```diff
@@ -144,7 +144,7 @@ def gnr_graph(n, p, create_using=None, seed=None):
     for source in range(1,n):
         target=random.randrange(0,source)
         if random.random() < p and target !=0:
-            target=G.successors(target)[0]
+            target = list(G.successors(target))[0]
```

Comment:
This should be changed to next(G.succssors(target))
to avoid creating the whole list just to get the first element.